r/tests/testthat/test-s3-minio.R

```diff
@@ -180,6 +190,20 @@ if (arrow_with_s3() && process_is_running("minio server")) {
       expect_length(fs$ls(minio_path("new_dataset_dir")), 1)
     })
 
+    test_that("CreateDir fails on bucket if allow_bucket_creation=False", {
+      now_tmp <- paste0(now, "-test-fail-delete")
+      fs$CreateDir(now_tmp)
+
+      expect_error(
+        limited_fs$CreateDir("should-fail"),
+        regexp = "Bucket does not exist"
+      )
+      expect_error(
+        limited_fs$DeleteDir(now_tmp),
+        regexp = "Would delete bucket"
+      )
+    })
```

Comment:
I wonder if these tests might be better without the `regexp` parameter, given that the specific phrasing of them comes from the C++ layer, and we tend to avoid testing that component of the error message?

Reply:
Sure, I'll rewrite it so these messages are tested in C++ instead.